test/xbgpu/test_engine.py

```diff
@@ -984,6 +1053,22 @@ def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.Heap
         delays[..., 0] = rng.uniform(-1e-9, 1e-9, size=(len(beam_outputs), n_ants))
         # Phase is in radians
         delays[..., 1] = rng.uniform(-2 * np.pi, 2 * np.pi, size=(len(beam_outputs), n_ants))
+
+        katcp_requests: dict[str, list[tuple[str, ...]]] = {
```

Comment:
The type hint `tuple[str, ...]` is wrong: it means a tuple containing only strings, whereas `weights` and `quant_gains` contain floats.

Reply:
Aha, I was under the impression the `...` type hint is similar to numpy indexing
* That is, "I'm not concerned with these dimensions"
* Slash, "They can be anything, as long as the first item/index is a string"

I'll figure something out. Was hesitant to use `typing.Any` as the requests don't all have the same number of args.

Reply:
I don't think the type system supports the idea of "first element is a string, and the remaining element can be anything". I think the best you'll manage is plain `tuple` with no specification of what's in the tuple.